cvcl/forms.py

```diff
@@ -23,3 +30,37 @@ def __init__(self, *args, **kwargs):
         user = kwargs.pop('user', None)  # pop the 'user' from kwargs dictionary
         super(VmDefinitionForm, self).__init__(*args, **kwargs)
         self.fields['image'] = ModelChoiceField(queryset=user.images.all())
+
+
+class csvForm(forms.Form):
+    data_file = forms.FileField()
+
+    def clean_data_file(self):
+        f = self.cleaned_data['data_file']
+        if f:
+            ext = f.name.split('.')[-1]
+            if ext != 'csv':
+                raise forms.ValidationError('Please upload only csv file')
+
+        return f
+
+    def process_data(self, course_id):
+        f = io.TextIOWrapper(self.cleaned_data['data_file'].file)
+        reader = csv.DictReader(f)
+
+        for users in reader:
+
+            users = {re.sub("['\"]", "", str(k.encode('ascii', 'ignore')))[1:]: v for k, v in users.items()}
+
+            username = users['Username']
+            email = users['Username'] + getattr(settings, "EMAIL_DOMAIN", '@email.com')
```

Comment:
The default here doesn't make any sense. Empty string would be better than sending it to a known arbitrary domain.